linter.py

```diff
@@ -1,42 +1,46 @@
-#
-# linter.py
-# Linter for SublimeLinter3, a code checking framework for Sublime Text 3
-#
-# Written by Jon Surrell
-# Copyright (c) 2014 Jon Surrell
-#
-# License: MIT
-#
-
-"""This module exports the Gotype plugin class."""
-
+import os
 from os.path import dirname
+import logging
+
 from SublimeLinter.lint import Linter, util
 
 
-class Gotype(Linter):
-    """Provides an interface to gotype."""
+logger = logging.getLogger(__name__)
+
 
-    syntax = ('go', 'gosublime-go')
+class GotypeLinter(Linter):
     executable = "gotype"
     regex = r'(?P<filename>^.+):(?P<line>\d+):(?P<col>\d+):\s+(?P<message>.+)'
-    error_stream = util.STREAM_STDERR
-
-    def __init__(self, view, syntax):
-        """Initialize and load GOPATH from settings if present."""
-        super(Gotype, self).__init__(view, syntax)
-
-        gopath = self.get_view_settings().get('gopath')
-        if gopath:
-            if self.env:
-                self.env['GOPATH'] = gopath
-            else:
-                self.env = {'GOPATH': gopath}
+    defaults = {
+        'selector': 'source.go'
+    }
 
     def cmd(self):
         """Return the command line to run."""
         return [self.executable_path, "-e", dirname(self.filename)]
```

Comment:
No, cmd is now static like `cmd = ['gotype', '-e', '$file_path']`

`executable_path` and `executable` are deprecated.

Why overriding `communicate`?